### PR TITLE
[#1396] cmd: Allow to autogenerate documentation for all commands

### LINKED ISSUE #1396: Add doc generation command to all our CLIs
We need to add a command for generating CLI documentation in Markdown format or using a template specified.

Something like:
```
$ neofs-cli gendoc <outdir> [custom.template]
```

Related info: 
- https://github.com/spf13/cobra/blob/master/doc/md_docs.md
- https://github.com/spf13/cobra/tree/master/doc

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1396.
Signed-off-by: Evgenii Stratonikov <evgeniy@nspcc.ru>

### CODE REVIEW

File: pkg/util/gendoc/gendoc.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+	return visit(rootCmd, outDir, ext, depth, tm)
+}
+
+func visit(rootCmd *cobra.Command, outDir string, ext string, depth int, tm *template.Template) error {
+	if depth == 0 {
+		name := strings.Join(fullUse(rootCmd), "-")
+		name = strings.TrimSpace(name)
+		name = strings.ReplaceAll(name, " ", "-")
+		name = filepath.Join(outDir, name) + ext
```
Comment (cthulhu-rider) [inline pkg/util/gendoc/gendoc.go:112]: Will `ext` be separated by some symbol (e.g. `-` or `.`)?

File: pkg/util/gendoc/gendoc.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+			case gendocMarkdown:
+				return doc.GenMarkdownTree(rootCmd, args[0])
+			case gendocMan:
+				hdr := &doc.GenManHeader{
+					Section: "1",
+					Source:  "NSPCC & Morphbits",
+				}
+				return doc.GenManTree(rootCmd, hdr, args[0])
+			default:
+				return fmt.Errorf("type must be 'md' or 'man'")
```
Comment (cthulhu-rider) [inline pkg/util/gendoc/gendoc.go:74]: Use constants or leave `errors.New`.

File: pkg/util/gendoc/gendoc.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+			}
+		},
+	}
+
+	ff := gendocCmd.Flags()
+	ff.StringP(gendocTypeFlag, "t", gendocMarkdown, "type for the documentation ('md' or 'man')")
```
Comment (cthulhu-rider) [inline pkg/util/gendoc/gendoc.go:80]: Consider using constants.

**Review by cthulhu-rider — APPROVED** [review]

File: pkg/util/gendoc/gendoc.go (hunk 1 above)
Comment (fyrchik) [reply pkg/util/gendoc/gendoc.go:112]: No, it should be included in the argument.

File: pkg/util/gendoc/gendoc.go (hunk 2 above)
Comment (fyrchik) [reply pkg/util/gendoc/gendoc.go:74]: Fixed.

File: pkg/util/gendoc/gendoc.go (hunk 3 above)
Comment (fyrchik) [reply pkg/util/gendoc/gendoc.go:80]: We have 2 and they are small, let's leave it compact.

File: pkg/util/gendoc/gendoc.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+The last optional argument specifies the template to use with text/template.
+In this case there is a number of helper functions which can be used:
+  replace STR FROM TO -- same as strings.ReplaceAll
+  join ARRAY SEPARATOR -- same as strings.Join
+  split STR SEPARATOR -- same as strings.Split
+  fullUse CMD -- slice of all command names starting from the parent
+  listFlags CMD -- list of command flags
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (carpawell) [inline pkg/util/gendoc/gendoc.go:43]: usually we tried not to use `RunE` in CLIs: https://github.com/nspcc-dev/neofs-node/issues/623

Comment (carpawell) [reply pkg/util/gendoc/gendoc.go:43]: created [issue](https://github.com/nspcc-dev/neofs-node/issues/1409)

**Review by carpawell — DISMISSED** [review]

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**